tesp/sensitivity.py

```diff
@@ -0,0 +1,135 @@
+import os
+import re
+import csv
+import shutil
+from os.path import join as pjoin, basename, exists
+
+import luigi
+import h5py
+import numpy
+import rasterio
+
+from wagl.acquisition import acquisitions
+from wagl.singlefile_workflow import DataStandardisation
+from tesp.package import PATTERN2, ARD
+from tesp.workflow import RunFmask
+
+
+def aerosol_summary(l2_path, fmask_path, granule, aerosol):
+
+    with rasterio.open(fmask_path) as mask_file:
+        fmask = mask_file.read(1)
+
+    valid_pixels = fmask == 1
+
+    def mask_invalid(img):
+        return numpy.where(img != -999, img, numpy.nan)
+
+    yield ['product', 'date', 'granule', 'band', 'aerosol', 'mean', 'std', 'valid_pixels']
+
+    with h5py.File(l2_path) as h5:
+
+        def get(*keys):
+            return h5['/'.join([granule, *keys])]
+
+        def band_dataset(product, band):
+            for res_group in ['RES-GROUP-2', 'RES-GROUP-1', 'RES-GROUP-0']:
```

Comment:
for any of the hdf5 offsets is it worthwhile using the format strings from wagl.constants? 

That way if we change them it'll be less work to migrate stuff across?

Reply:
Could you please have a look now? BTW I would like a squash merge for this one when its ready.